cmd/kube-apiserver/app/server.go

```diff
@@ -299,7 +299,18 @@ func Run(s *options.ServerRunOptions) error {
 	}
 
 	admissionControlPluginNames := strings.Split(s.GenericServerRunOptions.AdmissionControl, ",")
-	pluginInitializer := kubeadmission.NewPluginInitializer(client, sharedInformers, apiAuthorizer)
+	var cloudConfigReader *os.File
+
+	if s.CloudProvider.CloudConfigFile != "" {
+		cloudConfigReader, err = os.Open(s.CloudProvider.CloudConfigFile)
+		if err != nil {
+			glog.Errorf("Error opening cloud configuration file %s: %#v", s.CloudProvider.CloudConfigFile, err)
```

Comment:
Okay but api-server already seems to have a contract that, "I will not panic if I can't configure Cloudprovider properly". See - https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-apiserver/app/server.go#L158

If we were to change that contract, I do not want to do this as part of this particular fix.  We have already seen plenty of head scratching bugs where kubernetes wasn't working properly because cloudprovider wasn't configured properly and rather than complaining loudly kubernetes was silently logging error messages. So I completely agree with the idea of logging a fatal error, but I think we should do in both places not just here.

I am bit hesitant to change logging in both places though. I don't know enough about first initialization and why we aren't logging that as fatal error.